src/main/java/org/bptlab/cepta/DataToDatabase.java

```diff
@@ -0,0 +1,159 @@
+package org.bptlab.cepta;
+
+import com.github.jasync.sql.db.ConnectionPoolConfigurationBuilder;
+import com.github.jasync.sql.db.QueryResult;
+import com.github.jasync.sql.db.pool.ConnectionPool;
+import com.github.jasync.sql.db.postgresql.PostgreSQLConnection;
+import com.github.jasync.sql.db.postgresql.PostgreSQLConnectionBuilder;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.apache.flink.api.common.functions.MapFunction;
+
+public class DataToDatabase<Object> implements MapFunction<Object, Object> {
+  static final String HOST = "localhost";
+  static final String DATABASE = "dbs1_imdb";
+  static final String USER = "tester";
+  static final String PASSWORD = "password";
+  static final Integer PORT = 5432;
+  private String table_name;
+  public DataToDatabase(String table){
+    this.table_name = table;
+  }
+  @Override
+  public Object map(Object dataSet) throws Exception {
+    insert(dataSet);
+    return dataSet;
+  }
+  public void insert(Object dataSet)
+      throws NoSuchFieldException, IllegalAccessException {
+
+    // Connection to PostgreSQL DB
+    ConnectionPool<PostgreSQLConnection> connection;
+
+    ConnectionPoolConfigurationBuilder config = new ConnectionPoolConfigurationBuilder();
+    config.setUsername(USER);
+    config.setPassword(PASSWORD);
+    config.setHost(HOST);
+    config.setPort(PORT);
+    config.setDatabase(DATABASE);
+    config.setMaxActiveConnections(100);
+    connection = PostgreSQLConnectionBuilder.createConnectionPool(config);
+
+    // stores strings of values and columns for sql query
+    String[] colsAndVals = columnsAndValuesToString(dataSet);
+
+    // Create query
+    String query = "INSERT INTO " + table_name + colsAndVals[0]
+        + " VALUES " + colsAndVals[1] + ";";
```

Comment:
I feel like storing the column names and their values in the same array is fairly hard to understand and maintain and perhaps not necessary.

Reply:
That is because we only add columns to the string that actually have a value and are not null. If we separate those methods we have nearly the same code and work twice.